src/parsing.ts

```diff
@@ -76,30 +90,94 @@ export interface IParsedNode {
   children: IParsedNode[];
 }
 
-export const parseSource = (text: string) => {
+/**
+ * Look for test function imports in this AST node
+ *
+ * @param folder The workspace folder this file belongs to, used for relative path references to a custom test function
+ * @param fileUri The URI of the file we are extracting from
+ * @param testFunctions the tests function imports to check for
+ * @param node the ImportDelcaration to look for test imports
+ * @returns
+ */
+function importDeclarationExtractTests(
+  folder: vscode.WorkspaceFolder | undefined,
+  fileUri: vscode.Uri | undefined,
+  testFunctions: TestFunctionSpecifierConfig[],
+  node: ImportDeclaration,
+): ExtractTest[] {
+  const idTests: ExtractTest[] = [];
+  if (typeof node.source.value !== "string") {
+    return [];
+  }
+
+  let importValue = node.source.value;
+  if (node.source.value.startsWith("./") || node.source.value.startsWith("../")) {
+    if (!folder || !fileUri) {
+      console.warn(`Trying to match custom test function without specifying a folder or fileUri`);
+      return [];
+    }
+
+    // This is a relative import, we need to adjust the import value for matching purposes
+    const importRelativeToRoot = Path.relative(
+      folder.uri.fsPath,
+      Path.resolve(Path.dirname(fileUri.fsPath), node.source.value),
+    );
+
+    importValue = `./${importRelativeToRoot}`;
+  }
+
+  for (const specifier of testFunctions) {
+    if (specifier.import !== importValue) {
+      continue;
+    }
+
+    // Next check to see if the functions imported are tests functions
+    const validNames = new Set(
+      typeof specifier.name === "string" ? [specifier.name] : specifier.name,
+    );
+
+    for (const spec of node.specifiers) {
+      const specType = spec.type;
+      if (specType === C.ImportDefaultSpecifier || specType === C.ImportNamespaceSpecifier) {
+        if (validNames.has("default")) {
```

Comment:
The filter shouldn't happen at this level. This covers cases like `import * as t from 'node:test'`. We should then match validNames on accesses like `t.test` or `t.suite` (where the lower branch covers the cases like `import { test } from 'node:test`)

The unit tests in `parsing.test.ts` can help you. Run with `npx vitest`

Reply:
I think I didn't understand this comment fully until now. But I believe this is solved now.